geoindicators/src/main/groovy/org/orbisgis/SpatialUnits.groovy

```diff
@@ -177,25 +179,27 @@ static IProcess prepareRSUData(){
  * @param distance A distance to group the geometries
  * @param prefixName A prefix used to name the output table
  * @param outputTableName The name of the output table
- * @return A database table name.
+ * @return A database table name and the name of the column ID
  */
 static IProcess createBlocks(){
     return processFactory.create("Merge the geometries that touch each other",
             [inputTableName: String, distance : double, prefixName: String, datasource: JdbcDataSource],
-            [outputTableName : String],
+            [outputTableName : String, outputIdBlock: String],
             { inputTableName,distance =0.0, prefixName="block", datasource ->
                 logger.info("Merging the geometries...")
 
+                def columnIdName = "id_block"
+
                 // The name of the outputTableName is constructed
                 String baseName = "created_blocks"
                 String outputTableName = prefixName + "_" + baseName
 
                 datasource.execute "DROP TABLE IF EXISTS $outputTableName".toString()
-                datasource.execute "CREATE TABLE $outputTableName as  select  EXPLOD_ID as id_block, the_geom ".toString()+
+                datasource.execute "CREATE TABLE $outputTableName as  select  EXPLOD_ID as $columnIdName, the_geom ".toString()+
                         "from st_explode ('(select ST_UNION(ST_ACCUM(ST_BUFFER(THE_GEOM,$distance))) as the_geom".toString()+
                         " from $inputTableName)')".toString()
                 logger.info("The geometries have been merged")
-                [outputTableName: outputTableName]
+                [outputTableName: outputTableName, outputIdBlock: columnIdName]
```

Comment:
idem.
You are able to obtain it with the table metadata ?

Reply:
may be but I need to map it as explained above